src/Microsoft.DotNet.GenAPI/Shared/CSharpBuilder.cs

```diff
@@ -0,0 +1,218 @@
+    {
+        _syntaxWriter.WriteProperty(ps.ToDisplayString(AssemblySymbolDisplayFormats.MemberDisplayFormat),
+            hasImplementation: !ps.IsAbstract, ps.GetMethod != null, ps.SetMethod != null);
+    }
+
+    private void Process(IEventSymbol es)
+    {
+        _syntaxWriter.WriteEvent(es.ToDisplayString(AssemblySymbolDisplayFormats.MemberDisplayFormat),
+            es.AddMethod != null, es.RemoveMethod != null);
+    }
+
+    private void Process(IMethodSymbol ms)
+    {
+        _syntaxWriter.Writemethod(ms.ToDisplayString(AssemblySymbolDisplayFormats.MemberDisplayFormat),
+            hasImplementation: !ms.IsAbstract);
+    }
+
+    private IEnumerable<SyntaxKind> BuildAccessibility(ISymbol symbol)
+    {
+        switch (symbol.DeclaredAccessibility)
+        {
+            case Accessibility.Private:
+                return new SyntaxKind[] { SyntaxKind.PrivateKeyword };
+            case Accessibility.Internal:
+                return new SyntaxKind[] { SyntaxKind.InternalKeyword };
+            case Accessibility.ProtectedAndInternal:
+                return new SyntaxKind[] { SyntaxKind.PrivateKeyword, SyntaxKind.ProtectedKeyword };
+            case Accessibility.Protected:
+                return new SyntaxKind[] { SyntaxKind.ProtectedKeyword };
+            case Accessibility.ProtectedOrInternal:
+                return new SyntaxKind[] { SyntaxKind.ProtectedKeyword, SyntaxKind.InternalKeyword };
+            case Accessibility.Public:
+                return new SyntaxKind[] { SyntaxKind.PublicKeyword };
+            default:
+                throw new Exception(String.Format("Unexpected accesibility found {0}",
+                    SyntaxFacts.GetText(symbol.DeclaredAccessibility)));
+        }
+    }
+
+    private void BuildBaseTypes(INamedTypeSymbol namedType, ref List<string> baseTypeNames)
+    {
+        if (namedType.BaseType != null && namedType.BaseType.SpecialType == SpecialType.None &&
+            Filter.Include(namedType.BaseType))
+        {
+            baseTypeNames.Add(namedType.BaseType.ToDisplayString());
+        }
+
+        foreach (var interfaceSymbol in namedType.Interfaces)
+        {
+            if (!Filter.Include(interfaceSymbol)) continue;
+
+            baseTypeNames.Add(interfaceSymbol.ToDisplayString());
+        }
+    }
+
+    private void BuildConstrains(INamedTypeSymbol namedType, ref List<List<SymbolDisplayPart>> constrains)
+    {
+        bool whereKeywordFound = false;
+        var currConstrain = new List<SymbolDisplayPart>();
+
+        foreach (var part in namedType.ToDisplayParts(AssemblySymbolDisplayFormats.BaseTypeDisplayFormat))
+        {
+            if (part.Kind == SymbolDisplayPartKind.Keyword &&
+                part.ToString() == SyntaxFacts.GetText(SyntaxKind.WhereKeyword))
+            {
+                if (whereKeywordFound)
+                {
+                    constrains.Add(currConstrain);
+                    currConstrain = new List<SymbolDisplayPart>();
```

Comment:
Maybe you can do this?
```suggestion
                    currConstrain.Clear();
```

Reply:
Might have been some shennaginans with the `ref`? Please try again, this should just work